readme.md

```diff
@@ -45,3 +45,23 @@
 3. Add a `index.js` file with the plugin code
 4. Add a `readme.md` explaining what the plugin does, how to install, and how to configure it
 5. Submit a pull request
+
+### How to compose multiple plugins
+Probably you are going to use multiple plugins for your project such as Typescript with CSS. In this example using official next-typescript with next-css. 
+ :warning: some official and community plugins may not compose such as this example. some plugins have different input arguments and structure. so If It is not working like this example, please check each repos source code.
```

Comment:
It's probably worth adding a note about the order in which plugins could/should be composed together.

It's perhaps an attribute that could be specified in the plugin itself to be sure things don't break when used in a particular order.

I remember running into some issues of this sort when chaining together babel plugins.

Reply:
hm.... I want to add a note about it but I found that there are a lot of plugins, I don't know which ones are compatible and with specific order. 
I think the fundamental problem is that there is no standard on composing multiple plugins. 

**Problem  1 - There can be a conflict on options naming. if you see this basic example...**
```
module.exports = withTypescript(withCSS({
  cssModules: true,
  cssLoaderOptions: {
    importLoaders: 1,
  },
  typescriptLoaderOptions: {
    transpileOnly: false
  }
}));
```
Those naming of "cssLoaderOptions" "typescriptLoaderOptions"  is declared in each plugin and used internally to get options, but now config is in this big object with all the options for all composed plugins. It can cause a naming conflict between plugins.

**Problem 2 - About ordering problem**
I tested with very basic composing scenario and I didn't run into this problem. But you said you had some issues with this. Then the fact that this ordering problem exists is the problem... With current architecture It will likely be happening at anytime. I think It can be fixed with new architecture.

**Possible solution - completely isolate each plugins configuration**
 As you mentioned, https://github.com/JerryCauser/next-compose looks good to me. It isolates every plugin and its config with array. Also structure looks better to me comparing current HOC style structure

These are my thoughts...  I think there should be an architecture decision time about this matter